# Identification Secretory Peptides
This repository is part of the paper: ***"In silico identification of novel secretory peptides from bacteria proteome"***


# Computational Analysis of Signal Peptides Interacting with SecA in Escherichia coli

## Project Description
This repository contains the computational pipeline and data used for the identification and analysis of novel signal peptides (SPs) that interact with the SecA protein in Escherichia coli, as detailed in our study. This research explores the use of various bioinformatic tools to predict and model interactions between SPs and the Sec-pathway, with a focus on enhancing the efficiency of recombinant protein secretion.

## Table of Contents

- [Data Description](#data-description)
- [Methodology](#methodology)
- [Citation](#citation)
- [License](#license)

## Data Description
This project utilizes the following datasets:

    - Raw Data: Proteome extracted from UniProt database.
    - Processed Data: Interactions and scores generated by HADDOCK and other tools.


## Methodology

The computational pipeline includes:

## 1.- Protein Sequence Analysis
    
   Input: Proteome extracted from UniProt database [rawdata](https://doi.org/10.5281/zenodo.10971817) 
   
   1.1 This [jupiter notebook](/notebooks/1.2_Proteome_analysis_Sec_pathway_positionformatting.ipynb) shows the steps to process the input file "1.1_Secretome_database_validated_EColi_GFFfile.gff". The algorithm allowed filtering only experimentally validated SP -bearing proteins, reported as involved in cell secretion by the Sec-pathway using Python.

   1.2 [Ouput file](/data/1.2_Input_for_uniprot_IDmapping.txt): Text file with a full dataset showing 141 proteins identified. Each entry was labeled with UniProt IDs and amino acids positions from SP. Every entry shows a protein with the format “P0AEG4[1-20]”; representing the protein ID and the SP amino acids positions in brackets.

   1.3 Signal peptides associated with toxins (GO:0003824) from gram-negative bacteria proteome (taxonomy: Proteobacteria) were retrived from UniProt database. Both reviewed and hypothetical protein sequences were selected- [1.3_toxin_sequences_INPUT.fasta](https://doi.org/10.5281/zenodo.10971817). 

```
    signalp6 --fastafile input.fasta --organism other --output_dir Results --format txt --mode fasta
```
   



Filtering and preparation of sequences.

    Molecular Docking: Using HADDOCK for interaction prediction.
    Post-analysis: Energy calculations and structural evaluations.


## Citation

If you use this data or the computational pipeline in your research, please cite as follows:

´´´
Author(s), (Year). Title of the study. Journal/Publisher, [DOI/link to the publication]
´´´

## License

This project is licensed under the MIT License - see the LICENSE file for details.


