# Identification Secretory Peptides
This repository is part of the paper: ***"In silico identification of novel secretory peptides from bacteria proteome"***


# Computational Analysis of Signal Peptides Interacting with SecA in *Escherichia coli*

## Project Description
This repository contains the computational pipeline and data used for the identification and analysis of novel signal peptides (SPs) that interact with the SecA protein in Escherichia coli, as detailed in our study. This research explores the use of various bioinformatic tools to predict and model interactions between SPs and the Sec-pathway, with a focus on enhancing the efficiency of recombinant protein secretion.

## Table of Contents

- [Data Description](#data-description)
- [Methodology](#methodology)
- [Citation](#citation)
- [License](#license)

## Data Description
This project utilizes the following datasets:

- Raw Data: Proteome extracted from UniProt database.

- Processed Data: Interactions and scores generated by HADDOCK and other tools.


## Methodology

The computational protocol / pipeline includes:

## 1. Protein Sequence Analysis

### Input
Proteome extracted from the UniProt database. Access the data [here](https://doi.org/10.5281/zenodo.10971817).

### 1.1 Protein Filtering Process
This [Jupyter notebook](/notebooks/1.2_Proteome_analysis_Sec_pathway_positionformatting.ipynb) demonstrates the steps to process the input file `1.1_Secretome_database_validated_EColi_GFFfile.gff`. The algorithm filters out only experimentally validated SP-bearing proteins, reported as involved in cell secretion via the Sec-pathway, using Python.

### 1.2 Output File
The resulting [output file](/data/1.2_Input_for_uniprot_IDmapping.txt) contains a dataset listing 141XX?? identified proteins. Each entry is labeled with UniProt IDs and the amino acid positions of SPs. The format for each entry is `P0AEG4[1-20]`, where `P0AEG4` is the protein ID and `[1-20]` indicates the SP amino acid positions in brackets.

Finally, the control SP database stored 145 sequences, experimentally validated from *Escherichia coli* proteome (UniProt). The constructed collection of SPs included the signal peptide PelB from *Erwinia carotovora* [Supplementary Table 1](/outputs/SupplementaryTable1_SignalPeptides_with_experimental_evidence.xlsx). The Control SP database contains a total of 146 sequences.


### 1.3 Retrieval and Analysis of Toxin Signal Peptides

The toxin SP collection from gram-negative microbes is now being created following this protocol:"

Signal peptides associated with toxins (GO:0003824) from the proteome of gram-negative bacteria (taxonomy: Proteobacteria) were retrieved from the UniProt database, including both reviewed and hypothetical protein sequences. Access the sequences [here](https://doi.org/10.5281/zenodo.10971817). Subsequently, these sequences were processed using SignalP to predict signal peptides:

```bash
signalp6 --fastafile [input.fasta] --organism other --output_dir [outputDir] --format txt --mode fasta
```
In this case:

```bash
signalp6 --fastafile [1.3_toxin_sequences_INPUT.fasta] --organism other --output_dir [Results_toxin/] --format txt --mode fasta
```


The output file 1.3_prediction_results_toxin.txt includes predictions from SignalP, formatted as shown:

```
sp|A0A0H3AIG7|VGRG1_VIBC3_Actin_cross-linking_toxin_VgrG1_OS=Vibrio_cholerae_se NO_SP 1.000057 0.000001 0.000000 0.000000 0.000000 0.000000
sp|A0A0H3NK84|SSEK1_SALTS_Protein-arginine_N-acetylglucosaminyltransferase_SseK NO_SP 1.000040 0.000000 0.000000 0.000000 0.000000 0.000000
sp|A0A1S4NYE3|CDIA_ECOST_tRNA_nuclease_CdiA_OS=Escherichia_coli_(strain_STEC_O3	SP	0.024348	0.974034	0.000681	0.000438	0.000238	0.000230	CS pos: 32-33. Pr: 0.8066
...
```

The output file [1.3_prediction_results_toxin.txt](https://doi.org/10.5281/zenodo.10971817) need to be processed to extract and format useful data for further analysis. The [Jupyter notebook](/notebooks/1.3_ToxinPeptideDatabase_Generation_fromSignalP.ipynb) was used for processing SignalP output.

From the prediction, [917 newly reported signal peptide](/outputs/SupplementaryTable2.xlsx) candidates were identified, most of which exhibit a high SignalP predicted probability.



## 2. Signal peptide modeling

In this step, AlphaFold was used on the LatchBio platform (https://latch.bio/) to predict the structure of signal peptides. However, AlphaFold can also be used locally or in any other setting that the researcher deems appropriate.

The access to the PDB files for the control SP and the 917 newly reported SP is available [here](https://doi.org/10.5281/zenodo.10971817).

### 2.1.	Secondary structure of signal peptides

The STRIDE algorithm was used to analyze the secondary structure signal peptide models. In the case of the SP group, we analyzed the 28,828 residues from 917 PDB models in our database. 

The output data is [here](https://doi.org/10.5281/zenodo.10971817).

**Protein modeling and features of toxin signal peptide database modeled by AlphaFold.**

![Fig2](/data/Fig2.JPG)

A) The structure of the signal peptides from A0A0M2KG05 and A0A2K8QM57 proteins is depicted. The accuracy predictions in each amino acid is colored according to the pLDDT score. Central regions of the peptide’s models exhibit a higher prediction accuracy (blue or sky-blue colored). In contrast, outer regions in yellow and oranga are characterized by a lower prediction accuracy. The central region shows alpha helix structures while the outer regions have a disordered linear structure. B) The plot depicts quality proportion from 28,828 residues from the toxin SP database in terms of pLDDT score (very high, confidence, low and very low) C) The secondary structures are shown (helix, coil, turn and beta), the plot represents ratios from 28,828 residues from toxin signal peptide database. 

## 3. SecA-Signal peptide complex prediction using HADDOCK

In this section, interactions between SecA and SP were individually analyzed using the HADDOCK web server, which generated 10 clusters with 4 docking models each.


### 3.1.	Selection of representative molecular docking models

 The protocol can be followed in this [Jupyter notebook](/notebooks/3.1_Compute_IRMSD_from_HADDOCKfiles.ipynb).

## 4.	Interface analysis of SecA/SP molecular docking models (with MAPIYA).




===

Filtering and preparation of sequences.

    Molecular Docking: Using HADDOCK for interaction prediction.
    Post-analysis: Energy calculations and structural evaluations.


## Citation

If you use this data or the computational pipeline in your research, please cite as follows:

´´´
Author(s), (Year). Title of the study. Journal/Publisher, [DOI/link to the publication]
´´´

## Contributing

Contributions to this repository are welcome. Please fork the repository and submit pull requests with your proposed changes.

## License

This project is licensed under the MIT License - see the LICENSE file for details.


